### ISSUE TITLE
geom_rect pdf display issue

### ISSUE THREAD

**Issue by jacobkasper** [tidyverse/ggplot2#5429]
### Tips for a helpful bug report:

* If you have a question, please don't use this form. Instead, ask on <https://stackoverflow.com/> or <https://community.rstudio.com/>.

* Please include a **minimal reproducible example**, a reprex, to demonstrate the bug.
If you've never heard of a reprex, please read ["Make a reprex"](https://www.tidyverse.org/help/#reprex).
Do not include session info unless it is explicitly asked for.

* If you can, use one of the built-in datasets or a small toy dataset that exposes the bug.
If for some reason, the bug only occurs on your original data, try to limit the number of rows that are necessary to expose the bug. 
Share such data by copying `dput()` output rather than an external file.

* Unless the bug is about the theme, labels, scales or other plot decoration: please omit these from the code.
    
* Please check whether somebody has reported the same problem in the [issues](https://github.com/tidyverse/ggplot2/issues).

Delete these instructions once you have read them.

---
When printing a pdf of a plot that includes geom_rect and alpha, there are overlapping edges of the rectangles creating a 'border.' This issue is not present when saving a .tiff.

```r
df <- data.frame(
    x = rep(c(2, 5, 7, 9, 12), 2),
    y = rep(c(1, 2), each = 5),
    z = factor(rep(1:5, each = 2)),
    w = rep(diff(c(0, 4, 6, 8, 10, 14)), 2)
)

pdf('tmp.pdf')
ggplot(df, aes(xmin = x - w / 2, xmax = x + w / 2, ymin = y, ymax = y + 1, fill = z)) +
    geom_rect(color = NA, alpha = 0.3) +
    coord_cartesian(expand = FALSE) +
    theme( panel.grid.major = element_blank(),
          panel.grid.minor = element_blank() )
dev.off()



```

**Comment by teunbrand** [comment]
For reference, this issue was also posted on [StackOverflow](https://stackoverflow.com/questions/77126918/geom-rect-borders-with-pdf-output). I essentially think that Andy Baxter's comment over there captures the issue:

> I think it's perhaps a display thing - viewing the pdf overlaps the semi-transparent boxes slightly and renders a tiny 'border'. Converting the pdf to svg has no border. Zooming in doesn't 'thicken' the border appearance. Taking off transparency makes any apparent border disappear. That would indicate the border isn't really "there". Not sure what the fix would be for displaying though.

There isn't any border defined in terms of graphical instructions that ggplot2's hands off to grid or even grid to pdf. If you open the pdf in e.g. inkscape or illustrator, the rectangles don't have any strokes. This suggests that the graphical properties are encoded as they should. Moreover, the same happens if you try a similar plot with `plot()`:

```r
library(ggplot2)

df <- data.frame(
  x = rep(c(2, 5, 7, 9, 12), 2),
  y = rep(c(1, 2), each = 5),
  z = factor(rep(1:5, each = 2)),
  w = rep(diff(c(0, 4, 6, 8, 10, 14)), 2)
)

pdf('tmp.pdf')
with(df, plot(x = range(x - w / 2, x + w / 2), y = range(y, y + 1), type = "n"))
for (i in seq_len(nrow(df))) {
  with(df[i, ], rect(
    xleft = x - w / 2, xright = x + w / 2,
    ybottom = y, ytop = y + 1,
    col = alpha(as.integer(z), 0.3), lty = 0
  ))
}
dev.off()
```

So to summarise, I don't think this is an issue with ggplot2 or even an R graphics issue, but rather an issue with the display of the pdf.

**Comment by teunbrand** [comment]
Closing this issue as this is unlikely controlled by ggplot2.
